WordPress/src/main/java/org/wordpress/android/ui/posts/PrepublishingHomeViewModel.kt

```diff
@@ -39,11 +41,22 @@ class PrepublishingHomeViewModel @Inject constructor(
     private val _onSubmitButtonClicked = MutableLiveData<Event<PublishPost>>()
     val onSubmitButtonClicked: LiveData<Event<PublishPost>> = _onSubmitButtonClicked
 
-    fun start(editPostRepository: EditPostRepository, site: SiteModel) {
+    fun start(
+        editPostRepository: EditPostRepository,
+        site: SiteModel,
+        isPublishSecondaryEditorAction: Boolean,
+        isNewPost: Boolean
+    ) {
         if (isStarted) return
         isStarted = true
 
-        setupHomeUiState(editPostRepository, site)
+        if (isPublishSecondaryEditorAction) {
```

Comment:
I'm not sure I follow this flow. SecondaryAction is an abstract term - how do we know that we need to "publishImmediately"? 

If we are just relying on the existing usage I'm not sure it's a good idea. All this code would ideally work (without any changes to the prepublishing nudges classes) even after someone decides to add "Save as draft, update, save, publish, schedule, ..." as a secondary action to the Editor. Wdyt?

Reply:
> I'm not sure I follow this flow. SecondaryAction is an abstract term - how do we know that we need to "publishImmediately"?

hey @malinajirka Thanks for asking. This is done by [a flag passed to the bottom sheet](https://github.com/wordpress-mobile/WordPress-Android/pull/12006/files#diff-38b6aea9cfff5f3e0e4b7389b44de602R1736). So the flag is only set in the `PUBLISH_NOW` secondary action in the `EditPostActivity`. I wasn't sure if the logic would be clear enough.
Let me know what you think. 
Another approach would be to pass in the `SecondaryAction` enum and using a use case to determine this.

Reply:
> If we are just relying on the existing usage I'm not sure it's a good idea. All this code would ideally work (without any changes to the prepublishing nudges classes) even after someone decides to add "Save as draft, update, save, publish, schedule, ..." as a secondary action to the Editor. Wdyt?

If they add those actions it wouldn't be triggered. But indeed the solution wasn't clear enough to make that known so I am open to suggestions in addition to what was mentioned above. Thanks 😄

Reply:
>If they add those actions it wouldn't be triggered. But indeed the solution wasn't clear enough to make that known so I am open to suggestions in addition to what was mentioned above. Thanks 😄

What do you mean it wouldn't get triggered? If I add "showPrepublishingNudgeBottomSheet(..)" [here](https://github.com/wordpress-mobile/WordPress-Android/blob/ac2b69651a99d04f1856e302f725e5abde659ffa/WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java#L1241) or [here](https://github.com/wordpress-mobile/WordPress-Android/blob/ac2b69651a99d04f1856e302f725e5abde659ffa/WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java#L1245), it'll get triggered right? I'll pass "isSecondaryAction" == true, since it's true that it's a secondary action. And the BottomSheet will perform "PublishNow". Even though the user pressed "Save/Submit.." and I correctly set "isSecondaryAction == true". Or Am I missing something?


If I'm missing something ignore the rest of this message.

I think we have two options here
1. Pass a specific action the user pressed - eg. using an enum.
2. Rename "isPublishSecondaryEditorAction" (even the flag) to "PublishNow".

The current solution is imo confusing. Because it seems like "publishSecondaryEditorAction" is always just "PublishNow" which is not true.

Reply:
Resolved by clearing the status before opening the sheet.